### ISSUE TITLE
Provide guidelines for documenting object names

### ISSUE THREAD

**Issue by daobrien** [StyleGuides/WritingStyleGuide#223]
This discussion started around the use "application streams" vs "AppStreams" and "ApplicationStreams" but extends to other objects and concepts such as "deployment configuration" and "DeploymentConfig."

Note: I use the term "object names" somewhat loosely because not all applications use the same terminology. Sometimes they are called resources, sometimes other things.
Feel free to add more examples (e.g., BuildConfig) to help clarify this.

We need to answer several questions:
1. When can we abbreviate and when not?
2. When must we capitalize and when not?
3. Which DocBook or other tags to use, and when.

**Comment by m-czernek** [comment]
If we are talking about `ConfigMap` vs `configuration map`, the later is objectively less precise. If you say `configuration map`, that can mean many things. If you say `ConfigMap`, that means precisely one thing in the context of Kubernetes. 

If CS asked me to change something like `DeploymentConfig` to `deployment config`, I'd reject that as changing the meaning of the text (I'd reject it in both lecture and GEs). I'd say that in `DeploymentConfig` capitalization is not a matter of style. If YAML uses this capitalization, so will I. If someone says oc uses it differently, then both capitalization styles are correct, and I'd pick one and be consistent. 

`AppStream` is a different matter, and I honestly don't care either way.

I just wanted to point out that if you're talking about `BuildConfig`, that's very different from a package name, or a tech name (such as `AppStream` vs `ApplicationStream` vs `AppStreams` and whatever else).

**Comment by daobrien** [comment]
Here's another example:
~~~
To increase the number of Placement Groups for a Ceph cluster:
~~~
Should "placement groups" be init-capped here?

**Comment by mweetman-redhat** [comment]
I'd say keep it lower.. it's never PlacementGroups, always 2 words

**Comment by daobrien** [comment]
> I'd say keep it lower.. it's never PlacementGroups, always 2 words

Yeah that's the way I'm leaning for this example, and product docs does as well. Just change the case to match the UI or commands, as required.

**Comment by fiona-allen** [comment]
In upstream documentation Placement Group is always capitalized because it is often shortened to PG in any following references.
For example, A Placement Group (PG) is a logical.........The PGs are essential for the scalability.......

Even in Red Hat documentation for Ceph we refer to them capitalized.
https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/1.3/html/storage_strategies_guide/placement_groups_pgs

**Comment by daobrien** [comment]
I went here: https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/4/html/storage_strategies_guide/placement_groups_pgs

Placement Groups: 7 matches
placement groups: 120 matches

**Comment by flozanorht** [comment]
I strongly disagree with Marek about Kubernetes resource type names. IMHO all "names of things" should be plain English words, not tagged, not capitalized.

First, because the keyworks for YAML files and oc commands require different capitalization.

Second, because using keywords require tagging and we wanna reduce tagging.

Third, because using real works such as "deployment configuration" reads better than invented words such as "DeploymentConfig". Sometimes common concepts (or domain-specific concepts and jargon) become keywords in IT but that does require us to assume they're always keywords. A counter example would be "class" from Java and C++. Sometimes mention the concept of the thing named "class", sometimes the keyword "class" from source code.

Fourth because some Kuberntes concepts are simple common words such as secrets and services. I think all should use the same treatment. I would not tag a "secret" but I have to tag a "DeloymentConfig". That affects translations: some cultures translate jargon, some do not. Keywords are never translated. Following the "class" example, pt-BR will talk about "classes" and everyone knows that it is the "class" construct in Java. Translators should handle jargon, not our tagging.

IMHO AppStrean, BuildConfig, class, and placement groups are instances of the same. We should not have different decisions for Ansible, Kubernetes, Ceph, Yum, whatever. We should have a consistent set of generic guidelines and avoid creating exceptions. Prose is not supposed to use keywords from source code and config file syntax except when you are really taking about "write this in the file", then you need and tag the proper keyword.

**Comment by m-czernek** [comment]
Yes, we disagree.

> First, because the keyworks for YAML files and oc commands require different capitalization.

As I said, both are correct then. Pick one and stick with it.

> Second, because using keywords require tagging and we wanna reduce tagging.

You want to reduce visual clutter, not tagging. Two different things. Tagging is useful for translation as well as for understanding the content.

> Third, because using real works such as "deployment configuration" reads better than invented words such as "DeploymentConfig".

Disagreed. It might read better if you're reading it out loud. It is, however, highly confusing if you say `deployment configuration` instead of `DeploymentConfig`. Deployment configuration can mean so many thing. Do you mean Jenkinsfile of the underlying CI/CD? Do you mean Dockerfile of the underlying code? Do you mean OCP configuration? Do you mean the actual deploymentconfig?

In essence, you're pushing the explicit information of `we are talking about DeploymentConfig` into the context, which makes is more difficult to understand.

In the end, I write text that as minimal and as precise as possible. If something has semantic meaning, it is not a matter of style. You simply cannot replace `deploymentconfig` or `DeploymentConfig` for `deployment configuration`.

**Comment by flozanorht** [comment]
Marek, I think you are taking the DeploymentConfig example as the general rule and mandating that these names of concepts cannot be translated. I thing these are the exception instead of the common scenario. I think that services, secrets, classes (from Java and C++) and similar should dictate the general rule and things similar to DeploymentConfig and AppStream should follow the rule for consistency. I wanna avoid dealing with special cases. 

You are correct that by "tagging" I meant "visual clutter". But if you do use a keyword in prose, such as DeplymentConfig, you need the tagging and get the clutter. I'd rather avoid these by avoiding keywords unless I am explicity giving instructions to use the keyword in a command, config file, or source code.

Everywhere else, that is when I am talking about the concept related to that keyword, I prefer to use plain English words, that is, jargon. Do we tag all jargon and all terms with special meaning in the context of our products and courses? Of course no! We do not tag things like "IP" and "DNS". And translators know how to handle them.

**Comment by m-czernek** [comment]
> Marek, I think you are taking the DeploymentConfig example as the general rule and mandating that these names of concepts (or "names of things") cannot be translated.

I am not mandating anything; on the contrary, I am asking for certain items to be left out of this "mandate". You are asking for one grand rule to be applied to everything.

> I wanna avoid dealing with special cases.

And I want to avoid one grand rule to rule them all, which then CS will try to enforce despite clearly not applying in that particular case. 

>  I think that services, secrets, classes (from Java and C++) and similar should dictate the general rule

The situation is identical here. If there's a `Service` as an OCP concept, that's very different from a service, which can be a microservice, a managed service, or whatever else.  You can omit tagging because it's a one-word concept. Incidentally, you can omit tagging DeploymentConfig in lecture as well, while preserving the exact semantics. 

> We do not tag things like "IP" and "DNS". And translators know how to handle them.

You are contradicting yourself. Instead `DeploymentConfig`, which is equivalent to `IP`, you want `deployment configuration`, which is the equivalent to `internet protocol`. I guarantee you that if you wrote `set the internet protocol address of your server to 168.0.1.24`, that would confuse translators a lot. 

If you don't want to tag DeploymentConfig, don't. I support that. If you're telling me it has to be written as `deployment configuraton`, I'm telling you that's nonsense.

In general, you are saying you want one rule for everything. I'm saying that's not possible, and that you can decide whatever capitalisation you want, but that you need to leave a space for specific edge cases.

**Comment by alexpdp7** [comment]
In my opinion, I'm not seeing to any reason here to deviate from what every specific product's documentation uses. If we have an issue with what the upstream product docs use, we should raise an issue with them and only establish opposing guidelines if we have a very good reason to do so with ample agreement.

**Comment by daobrien** [comment]
No updates to this over the last two weeks so I'm going to review again and come up with a draft, which I'll paste here. It will be along the lines of "follow upstream practices unless there is a specific reason not to" but I'll elaborate to hopefully make it clear for everyone. Thanks for all the input.

**Comment by daobrien** [comment]
It looks like I've been lazy and haven't gotten back to this. It's been a year :( It's too late to address for the next release (a couple of weeks away) but I'll make sure it's addressed for the next (6 months away).
